include/deal.II/multigrid/mg_constrained_dofs.h

```diff
@@ -146,13 +168,53 @@ MGConstrainedDoFs::initialize(const DoFHandler<dim,spacedim> &dof)
 {
   boundary_indices.clear();
   refinement_edge_indices.clear();
+  level_constraints.clear();
 
   const unsigned int nlevels = dof.get_triangulation().n_global_levels();
 
-  //At this point refinement_edge_indices is empty.
+  //At this point level_constraint and refinement_edge_indices are empty.
+  level_constraints.resize(nlevels);
   refinement_edge_indices.resize(nlevels);
   for (unsigned int l=0; l<nlevels; ++l)
-    refinement_edge_indices[l] = IndexSet(dof.n_dofs(l));
+    {
+      IndexSet relevant_dofs;
+      DoFTools::extract_locally_relevant_level_dofs(dof,l,relevant_dofs);
+      level_constraints[l].reinit(relevant_dofs);
```

Comment:
I'm a little bit confused. What do you mean by "starting cell"? We are basically looping over all faces of owned and ghost cells that lie on the periodic boundary on a level, so can there not be a situation where the periodic neighbor of a ghost cell would be artificial? And if the cell on the periodic boundary is level owned, then the other should be ghosted, correct? I'm not exactly sure what the error should be.